config_ipa.go

```diff
@@ -26,19 +26,50 @@
 package verkle
 
 import (
+	"encoding/hex"
 	"os"
 
 	"github.com/crate-crypto/go-ipa/ipa"
 )
 
+// emptyCodeHashPoint is a cached point that is used to represent an empty code hash.
+// This value is initialized once in GetConfig().
+var (
+	emptyCodeHashPoint           *Point
+	emptyCodeHashFirstHalfValue  Fr
+	emptyCodeHashSecondHalfValue Fr
+)
+
+const (
+	codeHashVectorPosition     = 3 // Defined by the spec.
+	emptyCodeHashFirstHalfIdx  = codeHashVectorPosition * 2
+	emptyCodeHashSecondHalfIdx = emptyCodeHashFirstHalfIdx + 1
+)
+
 type IPAConfig struct {
 	conf *ipa.IPAConfig
 }
 
 type Config = IPAConfig
 
 func (ipac *IPAConfig) CommitToPoly(poly []Fr, _ int) *Point {
+	containsEmptyCodeHash := len(poly) >= emptyCodeHashSecondHalfIdx &&
+		poly[emptyCodeHashFirstHalfIdx].Equal(&emptyCodeHashFirstHalfValue) &&
+		poly[emptyCodeHashSecondHalfIdx].Equal(&emptyCodeHashSecondHalfValue)
+
+	if containsEmptyCodeHash {
+		poly[emptyCodeHashFirstHalfIdx] = FrZero
+		poly[emptyCodeHashSecondHalfIdx] = FrZero
+	}
+
 	ret := ipac.conf.Commit(poly)
+
+	if containsEmptyCodeHash {
+		ret.Add(&ret, emptyCodeHashPoint)
+		poly[emptyCodeHashFirstHalfIdx] = emptyCodeHashFirstHalfValue
+		poly[emptyCodeHashSecondHalfIdx] = emptyCodeHashSecondHalfValue
+	}
```

Comment:
If that's the case (i.e: `containsEmptyCodeHash == true`), then we zero-out the appropriate slots, calculate the commitment (so those slots will be ignored thus avoiding computation), and after we get the result we add the cached point (L68). Note that we also restore the original values in L69 and L70; this is important to keep `CommitToPoly` not mutate what the caller sent in the `poly` parameter.